src/VisualStudio/Core/Def/Implementation/Workspace/SourceGeneratedFileManager.cs

```diff
@@ -164,44 +151,32 @@ public void NavigateToSourceGeneratedFile(SourceGeneratedDocument document, Text
 
         public bool TryGetGeneratedFileInformation(
             string filePath,
-            [NotNullWhen(true)] out DocumentId? documentId,
-            [NotNullWhen(true)] out Type? generatorType,
-            [NotNullWhen(true)] out string? generatedSourceHintName)
+            out SourceGeneratedDocumentIdentity identity)
         {
             _foregroundThreadAffintizedObject.AssertIsForeground();
 
-            documentId = null;
-            generatorType = null;
-            generatedSourceHintName = null;
+            identity = default;
 
             if (!filePath.StartsWith(_temporaryDirectory))
             {
                 return false;
             }
 
             var fileInfo = new FileInfo(filePath);
-            if (!Guid.TryParse(fileInfo.Directory.Name, out var guid) ||
-                !_directoryInfoOnDiskByContainingDirectoryId.TryGetValue(guid, out var directoryInfo))
-            {
-                return false;
-            }
-
-            documentId = directoryInfo.DocumentId;
-            generatorType = directoryInfo.GeneratorType;
-            generatedSourceHintName = fileInfo.Name;
-
-            return true;
+            return Guid.TryParse(fileInfo.Directory.Name, out var guid) &&
```

Comment:
what is going on here.  how do you know the directory is a guid?

Reply:
So the Visual Studio shell (today) isn't a fan of files that don't really exist; we always need a file name that at least _sometimes_ has to exist or we can't really open things. So how us opening the dynamic files works is we generate a temporary file on disk, and then tell the shell to open it. When the file is opened, we go "ah, that's our file!" and then wire up from there. We just always make the containing directory name of that a GUID.